python/graphstorm/model/lm_embed.py

```diff
@@ -266,9 +266,6 @@ def __init__(self,
             "language model configurations must be provided"
 
         self._lm_models = LMModels(g, node_lm_configs, num_train, lm_infer_batch_size)
-        assert len(self._lm_models.ntypes) == len(g.ntypes), \
```

Comment:
GSPureLMNodeInputLayer is designed for the case when the encoder layer only has lm models. If there is a ntype does not have a corresponding LM model, you should not use GSPureLMNodeInputLayer.

Reply:
why do we have GSPureLMNodeInputLayer? shouldn't we always use `GSLMNodeEncoderInputLayer`?